plugin/storage/integration/cassandra_test.go

```diff
@@ -81,28 +82,28 @@ func (s *CassandraStorageIntegration) initializeCassandraFactory(flags []string)
 	return f, nil
```

Comment:
```suggestion
	t.Cleanup(func() {
		require.NoError(t, f.Close())
	})
	return f, nil
```

Reply:
> this gets called on each subtest

do you mean Cleanup of the parent `t` gets call for each of the child `t`s? I would've thought they would be independent, and all actual tests are run as `t.Run()` on the parent `t`

Reply:
Yes - I think because `Cleanup` is called _within_ each subtest ([example](https://github.com/jaegertracing/jaeger/blob/ea53d8e6ce6e0c40197dd2cdc6ffadff152f78ac/plugin/storage/integration/integration.go#L150)).

Here is a simplified version of what is happening:

```golang
func TestCleanup(t *testing.T) {
	t.Log("Start of test")
	for _, test := range []string{"one", "two"} {
		t.Run(test, func(t *testing.T) {
			t.Cleanup(func() {
				t.Log("Running cleanup")
			})
			t.Log("Running test", test)
		})
	}
	t.Log("End of test")
}
```

```
--- PASS: TestDummy (0.00s)
=== RUN   TestCleanup
    example_test.go:6: Start of test
=== RUN   TestCleanup/one
    example_test.go:12: Running test one
    example_test.go:10: Running cleanup
=== RUN   TestCleanup/two
    example_test.go:12: Running test two
    example_test.go:10: Running cleanup
=== NAME  TestCleanup
    example_test.go:15: End of test
--- PASS: TestCleanup (0.00s)
    --- PASS: TestCleanup/one (0.00s)
    --- PASS: TestCleanup/two (0.00s)
```

Reply:
`t.Cleanup` is called in the context of `t` where it's added. If you add a cleanup at the top of your test, it will run only once, whereas the one that's called inside sub-test function is indeed called for each subtest

Reply:
🤦‍♂️ sorry I overlooked your original suggestion

Reply:
Address in https://github.com/jaegertracing/jaeger/pull/5311/commits/0970720bdd8b21c2c8f51876776aca8ca2e477a7